src/lib.rs

```diff
@@ -66,15 +68,32 @@ use {
   prometheus_metrics::serve_prometheus,
 };
 
+#[derive(Parser, Debug)]
+struct CmdArgs {
+  #[arg(long, default_value_t = false)]
+  /// if you start multiple lemmy server instances set this to true on all but one of them
+  disable_scheduled_tasks: bool,
+  /// set to false to disable the http server
+  #[arg(long, default_value_t = true, action=ArgAction::Set)]
+  http_server: bool,
+  /// set to false to disable the outgoing federation in this process
+  #[arg(long, default_value_t = true, action=ArgAction::Set)]
+  federate_activities: bool,
+  /// the index of this outgoing federation process. only useful if you want to split federation work into multiple servers.
+  /// the first process has number 1.
+  #[arg(long, default_value_t = 1)]
+  federate_process_index: i32,
+  /// how many outgoing federation processes you are starting in total. if set, make sure to set --activity-process-index differently for each.
+  #[arg(long, default_value_t = 1)]
+  federate_process_count: i32,
+}
 /// Max timeout for http requests
 pub(crate) const REQWEST_TIMEOUT: Duration = Duration::from_secs(10);
 
 /// Placing the main function in lib.rs allows other crates to import it and embed Lemmy
 pub async fn start_lemmy_server() -> Result<(), LemmyError> {
-  let args: Vec<String> = env::args().collect();
-
-  let scheduled_tasks_enabled = args.get(1) != Some(&"--disable-scheduled-tasks".to_string());
-
+  let args = CmdArgs::parse();
+  let scheduled_tasks_enabled = !args.disable_scheduled_tasks;
```

Comment:
I think the `start_lemmy_server` function ought to take the `CmdArgs` as an argument. Otherwise it becomes very awkward to embed the function in another application, which is the whole point of the function.

The args ought to be parsed in `main.rs`, not here. Probably all the fields should also be `pub` for the same reason.